### PR TITLE
Displaying data from db

### PR DESCRIPTION
**Pull request by Prakhar-commits**
Displaying Data from the Database

### CODE REVIEW

File: src/types/ResponseTypes.ts [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+export interface Test {
+  [key: string]: string | number | boolean | null;
+}
+
+export interface Timeline {
+  [key: string]: string | number | boolean | null;
+}
+
+export interface Welcome3 {
```
Comment (suryabulusu) [inline src/types/ResponseTypes.ts:13]: Welcome3 is not a good name. It doesn't seem relevant to what we are building

File: src/utils/forminputhandling.ts [hunk 2]
```diff
@@ -2,8 +2,10 @@ import { RowType } from "@/types/types";
 import { instance } from "./RootClient";
 
 // get data from the db
-async function getData(page = 1, limit = 5) {
-  const { data } = await instance.get(`/quiz?_page=${page}&_limit=${limit}`);
+async function getData(page: number, limit: number) {
+  const { data } = await instance.get(
+    `api/session?session_id_is_null=true&offset=${page}&limit=${limit}`
```
Comment (suryabulusu) [inline src/utils/forminputhandling.ts:7]: you will need two functions. one where session id is null, and one where you make request directly to `api/session?offset=${page}&limit=${limit}`

and then process results. 

also u need to display only quizzes for quiz creator part. that is, where platform variable is set to "quiz". for now, display those where platform variable is set to "meet" too. but eventually, we need a filter for this. 

when we expand quiz creator to general session creator, we will have another tab wherein we display only "meet" platform sessions

File: src/utils/rootclient.ts [hunk 3]
```diff
@@ -2,4 +2,5 @@ import axios from "axios";
 
 export const instance = axios.create({
   baseURL: process.env.NEXT_PUBLIC_BACKEND_URL,
```
Comment (suryabulusu) [inline src/utils/rootclient.ts:4]: change name of this file to RootClient.ts

change name of forminputhandling to FormInputHandling ts

maybe call this db url instead of backend url ..

File: src/components/displayTable/DataDisplay.tsx [hunk 4]
```diff
@@ -61,7 +57,9 @@ const DataDisplay: React.FC<DataDisplayProps> = ({ getData }) => {
           pageCount={Math.ceil(totalItems / itemsPerPage)}
           marginPagesDisplayed={2}
           pageRangeDisplayed={5}
```
Comment (suryabulusu) [inline src/components/displayTable/DataDisplay.tsx:59]: what is this hardcoded number 5

File: src/components/displayTable/DataDisplay.tsx [hunk 5]
```diff
@@ -61,7 +57,9 @@ const DataDisplay: React.FC<DataDisplayProps> = ({ getData }) => {
...
-          onPageChange={({ selected }) => setCurrentPage(selected)}
+          onPageChange={({ selected }) =>
+            setCurrentPage(selected * itemsPerPage)
```
Comment (suryabulusu) [inline src/components/displayTable/DataDisplay.tsx:61]: can u check through this logic? 

note that "offset" in the get request command is related to "id" number. 

lets say there are 100 IDs.
they need not always be number 1 to 100

they could look like (in ulta order) 100, 99, 96, 92, 91

so in the next page, you should request for previous five IDs starting from 91. and not (100 - 5 = 95)

File: src/components/displayTable/Row.tsx [hunk 6]
```diff
@@ -2,26 +2,40 @@ import React, { useState } from "react";
...
-      reportSchedule,
-      endTime,
-      startTime,
-      synced,
-      isEnabled,
-    },
+    meta_data,
+    start_time,
+    end_time,
+    repeat_schedule: reportSchedule,
```
[the remark this answers is not in the document]
Comment (Prakhar-commits) [reply src/components/displayTable/Row.tsx:13]: It is repeat_Schedule only, I've resolved it now

File: src/types/ResponseTypes.ts (hunk 1 above)
Comment (Prakhar-commits) [reply src/types/ResponseTypes.ts:13]: Changes to DbTypes

File: src/components/displayTable/DataDisplay.tsx (hunk 4 above)
Comment (Prakhar-commits) [reply src/components/displayTable/DataDisplay.tsx:59]: It displays the number of pages . for example if there are total 10 pages and you are 3rd page the UI will look something like 
`previous 1 2 3 4 5 ... next`

File: src/components/displayTable/DataDisplay.tsx (hunk 5 above)
Comment (Prakhar-commits) [reply src/components/displayTable/DataDisplay.tsx:61]: changed it , used Id as lastId and passed it to getData. resolved

File: src/utils/rootclient.ts (hunk 3 above)
Comment (Prakhar-commits) [reply src/utils/rootclient.ts:4]: The name of file was changed in last PR only , weird it is showing here with root client  .
